comments_blackbox_test.go

```diff
@@ -0,0 +1,201 @@
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	testsupport "github.com/almighty/almighty-core/test"
+	almtoken "github.com/almighty/almighty-core/token"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// a normal test function that will kick off TestSuiteComments
+func TestSuiteComments(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, new(CommentsSuite))
+}
+
+// ========== TestSuiteComments struct that implements SetupSuite, TearDownSuite, SetupTest, TearDownTest ==========
+type CommentsSuite struct {
+	suite.Suite
+	workitemCommentsCtrl app.WorkItemCommentsController
+	workitemCtrl         app.WorkitemController
+	defaultSvc           *goa.Service
+	userSvc              *goa.Service
+	userSvc2             *goa.Service
+	db                   *gorm.DB
+	pubKey               *rsa.PublicKey
+	priKey               *rsa.PrivateKey
+	clean                func()
+	commentId            uuid.UUID
+}
+
+func (s *CommentsSuite) SetupSuite() {
+	var err error
+	if err = configuration.Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+	s.db, err = gorm.Open("postgres", configuration.GetPostgresConfigString())
+	if err != nil {
+		panic("Failed to connect database: " + err.Error())
+	}
+	// default service (without auth)
+	s.defaultSvc = goa.New("Comments-service-test")
+	// user service (with auth)
+	s.pubKey, _ = almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	s.priKey, _ = almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	s.userSvc = testsupport.ServiceAsUser("CommentsSuite-Service", almtoken.NewManager(s.pubKey, s.priKey), account.TestIdentity)
+	s.userSvc2 = testsupport.ServiceAsUser("CommentsSuite-Service", almtoken.NewManager(s.pubKey, s.priKey), account.TestIdentity2)
+	// Make sure the database is populated with the correct types (e.g. bug etc.)
+	if configuration.GetPopulateCommonTypes() {
+		if err := models.Transactional(s.db, func(tx *gorm.DB) error {
+			return migration.PopulateCommonTypes(context.Background(), tx, workitem.NewWorkItemTypeRepository(tx))
+		}); err != nil {
+			panic(err.Error())
+		}
+	}
+	s.clean = cleaner.DeleteCreatedEntities(s.db)
+	s.workitemCtrl = NewWorkitemController(s.userSvc, gormapplication.NewGormDB(s.db))
+	s.workitemCommentsCtrl = NewWorkItemCommentsController(s.userSvc, gormapplication.NewGormDB(s.db))
+}
+
+func (s *CommentsSuite) TearDownSuite() {
+	s.clean()
+	if s.db != nil {
+		s.db.Close()
+	}
+}
+
+func (s *CommentsSuite) SetupTest() {
+	// create the workitem that will be used to perform the comment operations during the tests.
+	// given
+	createWorkitemPayload := app.CreateWorkitemPayload{
```

Comment:
Assuming this test scenario 'file' will contain more test cases than just the ones that use this data, would it make sense to split out the 'required setup' from SetupTest?

If the data is created via a function call instead of a SetupTest than it's up to the test case to decide what it's requirements are. e.g. How many other tests cases will use this exact data set?